lighthouse-core/lib/asset-saver.js

```diff
@@ -119,6 +120,10 @@ function prepareAssets(artifacts, results) {
         const traceData = Object.assign({}, trace);
         const html = screenshotDump(screenshots, results);
 
+        if (results && results.audits) {
+          const evts = new Metrics(traceData.traceEvents, results.audits).generateFakeEvents();
+          traceData.traceEvents.push(...evts);
```

Comment:
On one hand, yeah, it feels bad to alter the trace. It'd be nice to keep the objects separate.

On the other:

* I don't think we have a need to keep around the original trace. 
* We've sorted the trace in-place already (a few times).
* The memory consumption is pretty huge for these traces, so there's that advantage to augmenting the solo trace.